### DISCUSSION TITLE
Desktop Java Library?

### DISCUSSION THREAD

**Discussion by ns96** [ggerganov/ggwave#78] [General]
Hi, 
Are there any plans for a Desktop Java Library? I currently make use of minimodem for a desktop Java application that simulates digital music playback from vintage cassettes and reel to reel tape decks. However, I would like to move away from that program since it doesn't seem like it's being actively developed anymore and it's just a pain to compile and get working on Windows.  Only thing I am not sure on with ggwave is the bit rate? Currently, I am using a bit rate of 1200 for minimodem to get the time synch working good i.e. a second of audio translates into a second of data to correctly control the playback of the digital sound file.

Here are the links to the Java desktop and esp32 projects.
https://github.com/ns96/cassetteflowJava
https://github.com/ns96/cassetteflow

Video of software in action using old tape deck.
https://youtu.be/pvrnPoVNLMM

**Comment by ggerganov** [comment]
Hi, very interesting project!

The bit rate of ggwave is much lower than 1200 bps. It's somewhere around 100 bps. The reason is that the FSK modulation used in ggwave is mostly focused for robust air-gapped communicaiton.

If you are going to be encoding/decoding over a cable, you can bump the bit rate by reducing the error correction bytes and using a faster custom protocol. I can help you with that - just let me know what sort of data you are communicating.

At the moment, I don't have plans for a full Java support. However, you can use ggwave through JNI.
A sample usage in Java through JNI is demonstrated here:

https://github.com/ggerganov/ggwave-java

**Reply by ns96** [reply]
Thanks for the quick reply. Now knowing a bit more about the targeted use case for ggwave its' design makes more sense to me. I will take a look at the JNI approach and extending it's functionality to see if that's a workable approach.

Thanks again
